packages/tlschema/src/shapes/TLTextShape.ts

```diff
@@ -15,6 +15,7 @@ export interface TLTextShapeProps {
 	textAlign: TLDefaultTextAlignStyle
 	w: number
 	text: string
+	richText?: string
```

Comment:
I really want to push back on `text`/`richText` being separate fields here - it feels really error prone and adds a lot of complexity. I'd like us to have a single source of truth for any given bit of info about a shape. The whole "make impossible states unrepresentable" thing.

Some alternatives (in order of my personal preference lol):
1. **Remove the plain-text-only path for shapes that support rich text**. It adds a lot of complexity, and I'm not convinced we actually need it. I very much doubt that we would add it if we were implementing text as rich text from scratch. Feels sort of premature-optimization-y to me but willing to be convinced - i might be missing context.
2. **Detect plain-text-only tiptap docs.** If document.content has only one entry and it's a paragraph with no attrs/marks, it's plain text. Most code can work under the assumption it just has a tiptap doc (because it does!), and special cases can choose to optimize for plain text if needed.
3. **Store both in a single field**. If `text` was `string | TipTapDoc`, we could use a `typeof text === 'string'` check to see if it's plain text without having to manually remember to check both paths in future code.

Let's maybe discuss offline though and figure out from there?

Reply:
yes! this is the main thing mentioned offline to you and in the comment in the description. i don't like it either. 
i wanted to discuss this migration before i proceeded further — we're on the same page here!

Reply:
→ PR https://github.com/tldraw/tldraw/pull/5131